yaptide/celery/utils/progress/fluka_monitor.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+from datetime import datetime, timezone
+import logging
+from pathlib import Path
+import re
+import threading
+from typing import Iterator, Optional, Tuple
+from yaptide.batch.watcher import TIMEOUT_MATCH
+
+from yaptide.celery.utils.requests import send_task_update
+from yaptide.utils.enums import EntityState
+
+# templates for regex matching output from `<simulation>_<no>.out` file
+S_OK_OUT_INIT = re.compile(r"^ Total time used for initialization:")
+S_OK_OUT_START = re.compile(r"^1NUMBER OF BEAM")
+S_OK_OUT_IN_PROGRESS = re.compile(r"^ NEXT SEEDS:")
+S_OK_OUT_COLLECTED = re.compile(r"^ All cases handled by Feeder")
+S_OK_OUT_FIN = re.compile(r"^ \* ======(?:( )*?)End of FLUKA [\w\-.]* run (?:( )*?) ====== \*")
+
+# for larger number of particles values are shifted accordingly to fit all digits, e.g. for 1e+12 particles:
+# "              1          999999999999          999999999999         2.8598309E-04         1.0000000E+30                 0       "
+S_OK_OUT_PROGRESS_REMAINING_LINE_EXAMPLE = \
+    "     980000                 20000                 20000             1.8515483E-04         1.0000000E+30             44182       "
+
+logger = logging.getLogger(__name__)
+
+
+def parse_progress_remaining_line(line: str) -> Optional[Tuple[int, int]]:
+    """Function parsing the line with progress remaining information.
+
+    Args:
+        line (str): line to be parsed
+    Returns:
+        Tuple[int, int]: tuple with two integers representing the progress and remaining.
+        If the line cannot be parsed None is returned.
+    """
+    list = line.split()
+    # expecting 6 sections which are int or floats
+    if len(list) != 6:
+        return None
+    try:
+        [float(x) for x in list]
+    except ValueError:
+        return None
+    return (int(list[0]), int(list[1]))
+
+
+@dataclass
+class TaskDetails:
+    simulation_id: int
+    task_id: str
+    update_key: str
+
+
+def time_now_utc() -> datetime:
+    """Function returning current time in UTC timezone."""
+    # because datetime.utcnow() is deprecated
+    return datetime.now(timezone.utc)
+
+
+def utc_without_offset(utc: datetime) -> str:
+    """Function returning current time in UTC timezone."""
+    return utc.strftime('%Y-%m-%d %H:%M:%S.%f')
+
+
+def read_fluka_out_file(event: threading.Event,
+                        linte_iterator: Iterator[str],
+                        next_backend_update_time: int,
+                        details: TaskDetails,
+                        varbose: bool = False) -> None:
+    """Function reading the fluka output file and raporting progress to the backend."""
+    in_progress = False
+    requested_primaries = 0
+    update_time = 0
+    for line in linte_iterator:
+        if event.is_set():
+            return
+        utc_now = time_now_utc()
+        if re.match(S_OK_OUT_START, line):
```

Comment:
```suggestion
        if re.match(S_OK_OUT_START, line):
```

I suspect that matching lines based on regexp may be costly. What about using `startstwith` method of a string ?
I think it should be worth to examine what is faster.

Another thing is that the order of regexp in the loop is important. If you have a line which you expect to be in the first part of the file and will appear only once, put it at the end of the loop. First should be the lines that you expect to be most frequent.